### PR TITLE
Support inline and widget links

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes https://github.com/exercism/v3-project-management/issues/142

- Support inline links
- Support widget links

### CODE REVIEW

**Comment by iHiD** [comment]
Fun!

File: app/commands/markdown/render_html.rb [hunk 1]
```diff
@@ -21,9 +21,19 @@ def link(node)
...
       out(' rel="nofollow"') if nofollow_links
+      out(link_tooltip_attributes(node))
       out('>', :children, '</a>')
     end
 
+    def link_tooltip_attributes(node)
+      link_match = %r{^(?<url>https://exercism.io)?/tracks/(?<track>[a-zA-Z0-9_-]+)/(?<type>concept|exercise)s/(?<slug>[a-zA-Z0-9_-]+)}.match(node.url) # rubocop:disable Layout/LineLength
+      return unless link_match
+
+      endpoint = Exercism::Routes.send("tooltip_track_#{link_match[:type]}_path", track_id: link_match[:track],
```
Comment (iHiD) [inline app/commands/markdown/render_html.rb]: 1. If we're going to arbitarily send methods like this, we need to ensure that `link_match[:type]` is an expected value, else we open ourselves up to injection. So I'd guard that it's either "exercise" or "concept" above this.
2. I don't **think** you need the `track_id` and `id` keys here. I think you can just pass the values

File: app/commands/markdown/render_html.rb [hunk 2]
```diff
@@ -21,9 +21,19 @@ def link(node)
       out('<a href="', node.url.nil? ? '' : escape_href(node.url), '" target="_blank"')
       out(' title="', escape_html(node.title), '"') if node.title.present?
       out(' rel="nofollow"') if nofollow_links
+      out(link_tooltip_attributes(node))
       out('>', :children, '</a>')
     end
 
+    def link_tooltip_attributes(node)
+      link_match = %r{^(?<url>https://exercism.io)?/tracks/(?<track>[a-zA-Z0-9_-]+)/(?<type>concept|exercise)s/(?<slug>[a-zA-Z0-9_-]+)}.match(node.url) # rubocop:disable Layout/LineLength
```
Comment (iHiD) [inline app/commands/markdown/render_html.rb]: Rather than `[a-zA-Z0-9_-]`, maybe it's safer to use `[^/]`. In case people put unicode in slugs etc (which I believe we don't disallow)

Comment (iHiD) [reply app/commands/markdown/render_html.rb]: (Maybe also add a test with a unicode char in the slug to be sure?)

Comment (ErikSchierboom) [reply app/commands/markdown/render_html.rb]: Good suggestion

File: app/commands/markdown/render_html.rb (hunk 1 above)
Comment (ErikSchierboom) [reply app/commands/markdown/render_html.rb]: Will do

File: app/commands/markdown/render_html.rb [hunk 3]
```diff
@@ -29,8 +29,7 @@ def link_tooltip_attributes(node)
       link_match = %r{^(?<url>https://exercism.io)?/tracks/(?<track>[a-zA-Z0-9_-]+)/(?<type>concept|exercise)s/(?<slug>[a-zA-Z0-9_-]+)}.match(node.url) # rubocop:disable Layout/LineLength
       return unless link_match
 
-      endpoint = Exercism::Routes.send("tooltip_track_#{link_match[:type]}_path", track_id: link_match[:track],
-                                                                                  id: link_match[:slug])
+      endpoint = Exercism::Routes.send("tooltip_track_#{link_match[:type]}_path", link_match[:track], link_match[:slug])
       " data-tooltip-type=\"#{link_match[:type]}\" data-endpoint=\"#{endpoint}\""
```
Comment (iHiD) [inline app/commands/markdown/render_html.rb]: Also, for this, it's better to use `%{...}` and not escape the `"`

File: test/commands/markdown/parse_test.rb [hunk 4]
```diff
@@ -128,4 +128,51 @@ class Markdown::ParseTest < ActiveSupport::TestCase
   test "removes html comments" do
     assert_equal "\n<p>Regular text</p>\n", Markdown::Parse.("<!-- Comment text -->\n\nRegular text\n", strip_h1: false)
   end
+
+  test "render internal concept link using absolute URL" do
+    expected = "<p><a href=\"https://exercism.io/tracks/ruby/concepts/basics\" target=\"_blank\" data-tooltip-type=\"concept\" data-endpoint=\"/tracks/ruby/concepts/basics/tooltip\">basics</a></p>\n" # rubocop:disable Layout/LineLength
```
Comment (iHiD) [inline test/commands/markdown/parse_test.rb]: Use `%{}` here too (etc to below)
